src/protocol/sort/apply.rs

```diff
@@ -0,0 +1,46 @@
+use embed_doc_image::embed_doc_image;
+/// Permutation reorders (1, 2, . . . , m) into (σ(1), σ(2), . . . , σ(m)).
+/// For example, if σ(1) = 2, σ(2) = 3, σ(3) = 1, and σ(4) = 0, an input (A, B, C, D) is reordered into (C, D, B, A) by σ.
+#[embed_doc_image("apply", "images/sort/apply.png")]
+#[allow(dead_code)]
+pub fn apply<T: Copy + Default>(sigma: &[usize], values: &[T]) -> Vec<T> {
```

Comment:
sigma is more scientific notation, in this context this seems to be permutation source indices and permutation dest indices for `apply_inv`

Reply:
Thanks for suggesting a better name!